packages/devtools_app/lib/src/memory/memory_controller.dart

```diff
@@ -1259,6 +1304,92 @@ class MemoryController extends DisposableController
     }
   }
 
+  void createDiffEnteries(Reference parent) {
+    for (final diffSnapshot in diffSnapshots) {
+      final diffSnapshotMatch = parent.children.firstWhere(
+        (element) {
+          var result = false;
+          if (element is DiffSnapshotReference) {
+            final DiffSnapshotReference node = element;
+            result = node.snapshot1 == diffSnapshot.snapshot1 &&
+                node.snapshot2 == diffSnapshot.snapshot2;
+          }
+
+          return result;
+        },
+        orElse: () => null,
+      );
+
+      if (diffSnapshotMatch == null) {
+        // New diff add it.
+        final diffSnapshotNode = DiffSnapshotReference(
+          diffSnapshot.snapshot1,
+          diffSnapshot.snapshot2,
+        );
+        parent.addChild(diffSnapshotNode);
+
+        final theDiffTree = buildDiffLibrary(diffSnapshot);
+        for (var index = 0; index < theDiffTree.length; index++) {
+          final libraryRef = theDiffTree[index];
+          diffSnapshotNode.addChild(libraryRef);
+        }
+        return;
+      }
+    }
+  }
+
+  static String formatDiffClassInfo(int totalNew, int totalDelete) =>
+      '+$totalNew(-$totalDelete)';
+
+  List<LibraryReference> buildDiffLibrary(SnapshotDiffing diffSnapshot) {
+    final result = <String, LibraryReference>{};
+
+    diffSnapshot.diffComputations.forEach((
+      HeapGraphClassLive key,
+      ListComputation<HeapGraphElementLive> value,
```

Comment:
@terrylucas. I'm having a bit of trouble understanding the diff computations. Is there a reason we need to have multiple diff computations vs having a single ahrd coded computation based on the identity hash code and perhaps field values to reduce false positives slightly?